### PR TITLE
feat: migration tool - seeding

### PR DESCRIPTION
**Pull request by lgw-gravitee**
To test create local K8s cluster or use remote
`kind create cluster`

Mongo:
`AM_LICENSE_PATH=/home/lgw/Downloads/gravitee-universe-v4.key ./scripts/migration-test.mjs --from-tag 4.10.15 --to-tag 4.11.8 --provider k8s  run`

Postgres:
`AM_LICENSE_PATH=/home/lgw/Downloads/gravitee-universe-v4.key ./scripts/migration-test.mjs --from-tag 4.10.15 --to-tag 4.11.8 --provider k8s --db-type postgres run`

You can apply --namespace param to override default `gravitee-am` 

Sample output
```
🛠️ Running Migration Orchestration
📝 Tags: 4.10.15 -> 4.11.8

▶️ Stage: clean
🧹 Cleaning up K8s environment...
🗑️  Uninstalling Helm releases...
🧹 Cleaning up MongoDB...
🗝️  Cleaning up secrets...
💀 Deleting namespace: gravitee-am...
✅ Cleanup complete.

▶️ Stage: k8s:setup
🏗️  Setting up K8s environment...
🍃 Deploying standalone MongoDB...

▶️ Stage: deploy-from
🚀 Deploying AM version 4.10.15...
🔌 Starting port-forward: svc/am-mapi-management-api 8093:83...
🔌 Starting port-forward: svc/am-mapi-management-ui 8002:8002...
🔌 Starting port-forward: svc/am-gateway-dp1-gateway 8091:82...
🔌 Starting port-forward: svc/am-gateway-dp2-gateway 8092:82...

▶️ Stage: seed-baseline

> gravitee-am-test@4.10.0-SNAPSHOT migration:seed
> cross-env TS_NODE_PROJECT=tsconfig.json TS_NODE_TRANSPILE_ONLY=1 node -r ts-node/register -r tsconfig-paths/register migration-seeding/bootstrap.ts --version 4.10 --label alpha


▶️ Stage: verify-all-alpha
🔍 Verifying alpha domain — full stack...
PASS specs/migration/gateway.jest.spec.ts (33.787 s)
  migration Gateway data [data plane dp1]
    ✓ logs in the user and redirects to an MFA step (2754 ms)
    ✓ logs in with MFA, creates a token and introspects it (3160 ms)
    ✓ grants consent for a requested scope and creates a token carrying it (1616 ms)
    ✓ logs in a user backed by the custom identity provider and creates a token (1731 ms)
    ✓ exchanges a JWT assertion for a token via the seeded jwt-bearer extension grant and introspects it (232 ms)
  migration Gateway data [data plane dp2]
    ✓ logs in the user and redirects to an MFA step (2633 ms)
    ✓ logs in with MFA, creates a token and introspects it (2700 ms)
    ✓ grants consent for a requested scope and creates a token carrying it (1287 ms)
    ✓ logs in a user backed by the custom identity provider and creates a token (1583 ms)
    ✓ exchanges a JWT assertion for a token via the seeded jwt-bearer extension grant and introspects it (214 ms)

PASS specs/migration/mapi-data.jest.spec.ts
  migration MAPI data [data plane dp1]
    ✓ keeps the versioned MAPI data readable through the generated SDK (560 ms)
  migration MAPI data [data plane dp2]
    ✓ keeps the versioned MAPI data readable through the generated SDK (369 ms)

Test Suites: 2 passed, 2 total
Tests:       12 passed, 12 total
Snapshots:   0 total
Time:        35.865 s
Ran all test suites matching /specs\/migration|specs\/migration/i.
⚠️  7 ERROR line(s) in am-mapi:
    [pod/am-mapi-management-api-8595bdccb-xvwdt/am-mapi-management-api] 10:15:23.109 [graviteeio-node] [] ERROR i.g.node.monitoring.metrics.Metrics - Gravitee metrics is disabled. You need to enable it first (services.metrics.enabled=true)
    [pod/am-mapi-management-api-8595bdccb-xvwdt/am-mapi-management-api] 10:15:29.100 [graviteeio-node] [] ERROR i.g.node.monitoring.metrics.Metrics - Gravitee metrics is disabled. You need to enable it first (services.metrics.enabled=true)
    [pod/am-mapi-management-api-8595bdccb-xvwdt/am-mapi-management-api] 10:15:36.695 [graviteeio-node] [] ERROR i.g.node.monitoring.metrics.Metrics - Gravitee metrics is disabled. You need to enable it first (services.metrics.enabled=true)
    [pod/am-mapi-management-api-8595bdccb-xvwdt/am-mapi-management-api] 10:15:37.288 [graviteeio-node] [] ERROR i.g.node.monitoring.metrics.Metrics - Gravitee metrics is disabled. You need to enable it first (services.metrics.enabled=true)
    [pod/am-mapi-management-api-8595bdccb-xvwdt/am-mapi-management-api] 10:15:37.593 [graviteeio-node] [] ERROR i.g.node.monitoring.metrics.Metrics - Gravitee metrics is disabled. You need to enable it first (services.metrics.enabled=true)
    [pod/am-mapi-management-api-8595bdccb-xvwdt/am-mapi-management-api] 10:18:05.507 [RxComputationThreadPool-1] [] ERROR i.g.node.monitoring.metrics.Metrics - Gravitee metrics is disabled. You need to enable it first (services.metrics.enabled=true)
    [pod/am-mapi-management-api-8595bdccb-xvwdt/am-mapi-management-api] 10:18:10.603 [RxComputationThreadPool-1] [] ERROR i.g.node.monitoring.metrics.Metrics - Gravitee metrics is disabled. You need to enable it first (services.metrics.enabled=true)
⚠️  58 ERROR line(s) in am-gateway-dp1:
    [pod/am-gateway-dp1-gateway-797786d994-8cr25/am-gateway-dp1-gateway] 10:16:46.420 [graviteeio-node] [] ERROR i.g.node.monitoring.metrics.Metrics - Gravitee metrics is disabled. You need to enable it first (services.metrics.enabled=true)
    [pod/am-gateway-dp1-gateway-797786d994-8cr25/am-gateway-dp1-gateway] 10:16:46.429 [graviteeio-node] [] ERROR i.g.node.monitoring.metrics.Metrics - Gravitee metrics is disabled. You need to enable it first (services.metrics.enabled=true)
    [pod/am-gateway-dp1-gateway-797786d994-8cr25/am-gateway-dp1-gateway] 10:16:46.430 [graviteeio-node] [] ERROR i.g.node.monitoring.metrics.Metrics - Gravitee metrics is disabled. You need to enable it first (services.metrics.enabled=true)
    [pod/am-gateway-dp1-gateway-797786d994-8cr25/am-gateway-dp1-gateway] 10:16:46.430 [graviteeio-node] [] ERROR i.g.node.monitoring.metrics.Metrics - Gravitee metrics is disabled. You need to enable it first (services.metrics.enabled=true)
    [pod/am-gateway-dp1-gateway-797786d994-8cr25/am-gateway-dp1-gateway] 10:16:46.430 [graviteeio-node] [] ERROR i.g.node.monitoring.metrics.Metrics - Gravitee metrics is disabled. You need to enable it first (services.metrics.enabled=true)

  migration MAPI data [data plane dp2]
    ✓ keeps the versioned MAPI data readable through the generated SDK (501 ms)

### CODE REVIEW

File: scripts/migration-tool/README.md [hunk 1]
```diff
@@ -28,21 +28,38 @@ All commands must be run from the **repository root** (e.g. `./scripts/migration
 
 ## Workflow and design
 
+**Terminology:** seeded data is keyed by a **channel label**, not by version, so the baseline and newline data sets stay distinct even when both sides share the same minor version (e.g. a `4.10.7 → 4.10.8` patch migration). The *baseline* channel (label **`alpha`**) is seeded with the **`--from-tag`** data and lives in domain `migration-seeded-domain-alpha`; the *newline* channel (label **`beta`**) is seeded with the **`--to-tag`** data and lives in domain `migration-seeded-domain-beta`. The `mapi`/`all` prefix on a verify stage only describes the system state at that point (Management API upgraded vs. full stack); the suffix (`baseline`/`newline`) picks which channel is asserted.
```
Comment (wajb) [inline scripts/migration-tool/README.md]: discussion: I have never encountered "newline" used in the sense of contrasting to a baseline 🤔. Is this an invention for this particular migration tool or would other readers be confused too? ("Target" was the term generally used previously, and "upgrade" was used in at least one command.)

"Alpha" and "beta" are also used too, hardcoded in places. I'm not sure if these are a different naming convention for the same thing.

File: scripts/migration-tool/README.md [hunk 2]
```diff
@@ -28,21 +28,38 @@ All commands must be run from the **repository root** (e.g. `./scripts/migration
...
+
+> The full pipeline requires a **concrete `--to-tag`** (e.g. `4.11.8`), not `latest`, because it seeds and verifies the new version. The channel asserted is derived automatically from the stage (baseline → `alpha`, newline → `beta`) — no `--test-label` needed.
```
Comment (wajb) [inline scripts/migration-tool/README.md]: question: what does "full pipeline" here mean? Also, if `latest` is not supported, is there any reason still to allow this value?

File: gravitee-am-test/api/commands/gateway/utils.ts [hunk 3]
```diff
@@ -22,7 +22,7 @@ export function getBase64BasicAuth(clientId: string, clientSecret: string) {
 }
 
 export function applicationBase64Token(app: Application) {
-  return getBase64BasicAuth(app.settings.oauth.clientId, app.settings.oauth.clientSecret);
+  return getBase64BasicAuth(app.settings.oauth.clientId, app.settings.oauth.clientId);
```
Comment (wajb) [inline gravitee-am-test/api/commands/gateway/utils.ts]: issue: not sure this makes sense for most usages. If it's needed for some scenarios, could the callers simply replace the secret?

File: scripts/migration-tool/env/k8s/db/db-mongodb.yaml [hunk 4]
```diff
@@ -11,9 +11,22 @@ auth:
   enabled: true
   # Bitnami chart key: K8s secret *name* only (no credentials in repo). Aqua false positive: not a password.
   existingSecret: "mongo-mongodb-auth"
-  # One user 'gravitee' with access to each dataplane DB (parallel arrays: usernames[i], passwords[i], databases[i])
-  usernames: ["gravitee", "gravitee", "gravitee"]
-  databases: ["gravitee", "gravitee-dp1", "gravitee-dp2"]
+  # Single user 'gravitee' (authSource = gravitee db). Multi-database access (gravitee-dp1/dp2)
+  # is granted via initdbScripts below: the Bitnami parallel-array form creates a username only
+  # once, so listing 'gravitee' three times did NOT grant it readWrite on the dataplane DBs
+  # (the dataplane gateways then failed with "not authorized on gravitee-dp1").
```
Comment (wajb) [inline scripts/migration-tool/env/k8s/db/db-mongodb.yaml]: nitpick: nice fix, but this explanation can be removed IMO.

File: scripts/migration-tool/lib/Orchestrator.mjs [hunk 5]
```diff
@@ -48,47 +48,70 @@ export class Orchestrator {
             case 'deploy-from':
                 await this.provider.deploy(this.options.fromTag);
                 break;
-            case 'verify-baseline':
-                await this.runTests('🔍 Running baseline tests...', 'ci:management:parallel', 'specs/management');
+            case 'seed-baseline':
+            case 'seed': // alias (backward compatible)
```
Comment (wajb) [inline scripts/migration-tool/lib/Orchestrator.mjs]: suggestion: we don't need to keep the old command.

File: scripts/migration-tool/lib/Orchestrator.mjs [hunk 6]
```diff
@@ -48,47 +48,70 @@ export class Orchestrator {
...
-            case 'verify-mapi':
-                await this.runTests('🔍 Verifying MAPI upgrade...', 'ci:management:parallel', 'specs/management');
-                break;
             case 'upgrade-gw':
                 await this.provider.upgradeGw(this.options.toTag);
                 break;
-            case 'verify-all':
-                await this.runTests('🔍 Final verification...', 'ci:gateway', 'specs/gateway');
+            case 'seed-newline':
+            case 'seed-upgrade': // alias (backward compatible)
```
Comment (wajb) [inline scripts/migration-tool/lib/Orchestrator.mjs]: suggestion: we don't need to keep the old command.

File: scripts/migration-tool/README.md [hunk 7]
```diff
@@ -167,33 +186,69 @@ Requires `CIRCLECI_TOKEN`. Sends parameters (from-tag, to-tag, db-type, provider
...
+The migration tool seeds deterministic data through the generated Management API SDK in `gravitee-am-test/api/management`.
+
+Seed files live in `gravitee-am-test/migration-seeding/versions/<major.minor>/seed.ts`. Each seed file must be idempotent and use stable identifiers, names, or metadata so migration specs can find the same objects after upgrade and downgrade. A seed file exposes `seed(label)` and names every entity by that **label** (not the version), so the same version's data can be seeded twice under different labels without colliding.
+
+The seed runner takes `--version <major.minor>` (which seed module / data shape to run) and `--label <name>` (the channel suffix used for naming). The `seed-baseline` stage runs the `--from-tag` module under label `alpha`; the `seed-newline` stage runs the `--to-tag` module under label `beta`. Because each channel targets a single version+label, this works for patch migrations too (e.g. `4.10.7 → 4.10.8` seeds the `4.10` module twice, as `alpha` and `beta`).
+
+Example local seed run against an already deployed Management API:
+
+```bash
+npm --prefix gravitee-am-test run migration:seed -- --version 4.10 --label alpha
```
Comment (wajb) [inline scripts/migration-tool/README.md]: question: why not use the script/orchestrator commands?

File: scripts/migration-tool/README.md [hunk 8]
```diff
@@ -167,33 +186,69 @@ Requires `CIRCLECI_TOKEN`. Sends parameters (from-tag, to-tag, db-type, provider
...
+
+Example migration test run against an already seeded environment:
+
+```bash
+AM_MIGRATION_TEST_LABEL=alpha npm --prefix gravitee-am-test run ci:migration
+```
+
+When running through the migration tool, `AM_MIGRATION_TEST_LABEL` is set automatically per stage (`alpha` for baseline stages, `beta` for newline stages; ad-hoc generic verify stages use `--test-label`, default `alpha`).
+
+Retrospective coverage for older releases is limited by the generated SDK and API contract available in the current branch. Prefer adding seeds before or during the release where the data shape is introduced. For older releases, keep seed data to API fields accepted by that older AM version.
```
Comment (wajb) [inline scripts/migration-tool/README.md]: discussion: this is a crucial limitation. We cannot reliably seed data for older versions using the current SDK, meaning we cannot target the edge cases the migration tests are really for.

The CI integration part will be key. Do you intend to sequence different AM checkouts in CircleCI for the seeding stages (our initial idea)? Or maybe there could be a way to embed a self-seeding command in (private?) mapi images?

If this is solved then we won't need to discourage covering behaviours across schema changes.

File: gravitee-am-test/migration-seeding/bootstrap.ts [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+
+export function toMinorVersion(version: string): string {
+  const match = version.match(/^(\d+)\.(\d+)/);
+  if (!match) {
+    throw new Error(`Invalid AM version: ${version}`);
+  }
+  return `${match[1]}.${match[2]}`;
+}
+
+export function applyDefaultSeedEnvironment(): void {
```
Comment (leleueri) [inline gravitee-am-test/migration-seeding/bootstrap.ts]: nickpick: my understanding is this method is using the env vers or the default if the var is not defined. Should id be named "applySeedEnvironment" instead?

File: scripts/migration-tool/README.md (hunk 1 above)
Comment (lgw-gravitee) [reply scripts/migration-tool/README.md]: I wanted to distinct data created before and after migration. The idea was 
- baseline (before upgrade) newline (after upgrade). They refer to created data.  
- alpha/beta are suffixes for domains (in case of upgrading the same major version 4.10.1 -> 4.10.2)

do you think the follwoing would be better?
domain-base / domain-new
and 
seed base/seed new?

File: gravitee-am-test/api/commands/gateway/utils.ts (hunk 3 above)
Comment (lgw-gravitee) [reply gravitee-am-test/api/commands/gateway/utils.ts]: thanks, I made it by a mistake

File: scripts/migration-tool/README.md (hunk 8 above)
Comment (lgw-gravitee) [reply scripts/migration-tool/README.md]: yes, I added checking out GIT worktress capability for seeding from specific version

File: scripts/migration-tool/README.md (hunk 7 above)
Comment (lgw-gravitee) [reply scripts/migration-tool/README.md]: I changed that.

File: scripts/migration-tool/README.md (hunk 2 above)
Comment (lgw-gravitee) [reply scripts/migration-tool/README.md]: I added support for 'latest'. it calculates the real version in the beginigng

File: gravitee-am-test/migration-seeding/bootstrap.ts (hunk 9 above)
Comment (lgw-gravitee) [reply gravitee-am-test/migration-seeding/bootstrap.ts]: I corrected to applyMissingSeedEnvironmentDefaults

File: .superpowers/sdd/final-fix-report.md
Comment (leleueri) [inline .superpowers/sdd/final-fix-report.md:1]: question: is it something we should commit ?

File: scripts/migration-tool/README.md (hunk 1 above)
Comment (wajb) [reply scripts/migration-tool/README.md]: Yes, I think that's clearer, thanks.

File: scripts/migration-tool/lib/Orchestrator.mjs [hunk 10]
```diff
@@ -106,6 +150,51 @@ export class Orchestrator {
...
+    /**
+     * Seed one version under a channel label. When a SeedWorktree is configured (and enabled), the
+     * seed runs from a worktree of `tag` so it uses that version's own SDK + scripts; otherwise (or
+     * when the tag predates the seeding framework) it falls back to the current checkout.
+     */
+    async seedStage(tag, label) {
+        const args = ['--version', toMinorVersion(tag), '--label', label];
+        let seedDir = null;
+        if (this.seedWorktree && this.options.seedFromWorktree) {
+            seedDir = await this.seedWorktree.resolveSeedDir(tag);
```
Comment (wajb) [inline scripts/migration-tool/lib/Orchestrator.mjs:167]: thought: it's still going to be difficult to maintain `gravitee-am-test/migration-seeding/versions/<version>` folders in parallel. API changes may easily break seeding of older versions, so, after fixing, what's left will always be identical for each version 🤔.

I wonder if only the worktree approach is valid and we would be better off with a single seeding folder that is free to differ between branches. Wdyt? Is there still value in keeping the seeding as stacked folders I'm overlooking?

File: gravitee-am-test/specs/migration/fixture/gateway-fixture.ts [hunk 11]
```diff
@@ -0,0 +1,294 @@
...
+    const finalAuthorizeResponse = await performGet(consentResponse.headers['location'], '', {
+      Cookie: consentResponse.headers['set-cookie'],
+    }).expect(302);
+
+    expect(finalAuthorizeResponse.headers['location']).toContain('code=');
+    return finalAuthorizeResponse;
+  }
+
+  async function loginAndCompleteMfa() {
+    await resetUserFactors();
```
Comment (leleueri) [inline gravitee-am-test/specs/migration/fixture/gateway-fixture.ts:170]: discussion: I understand why we are doing this resetUserFactor, this is to make sure that the TOTP shared secret is present with the right value. But by doing this we are not asserting that the existing facor instance is usable by the user, we always use a "fresh" enrolled factor.

This is fine for this PR but IMO in addition of this TOTP case, we should use an EMAIL or SMS factor and in the fixture function, we should adapt the execution based on the Location header value (if the header target enrollment endpoint, we enroll then challenge otherwise we are just challenging the user)

File: scripts/migration-tool/lib/Orchestrator.mjs (hunk 10 above)
Comment (leleueri) [reply scripts/migration-tool/lib/Orchestrator.mjs:167]: I tempt to agree with you @wajb , with the worktree option having dedicated seed folder per version may not be useful... 🤔 however wondering if this per version folder can make sens for the test cases because if we had a test case for agent introduced in 4.12 for example, those test cases make no sens for the seed version 4.11...

**Review by leleueri — APPROVED** [review]

File: scripts/migration-tool/lib/Orchestrator.mjs (hunk 10 above)
Comment (leleueri) [reply scripts/migration-tool/lib/Orchestrator.mjs:167]: we will discuss this later and iterate on it.

File: gravitee-am-test/specs/migration/fixture/gateway-fixture.ts (hunk 11 above)
Comment (leleueri) [reply gravitee-am-test/specs/migration/fixture/gateway-fixture.ts:170]: we will discuss this later and iterate
